diff --git a/jsk_perception/node_scripts/ssd_train_dataset.py b/jsk_perception/node_scripts/ssd_train_dataset.py
index 6c90d2d..9709de7 100755
--- a/jsk_perception/node_scripts/ssd_train_dataset.py
+++ b/jsk_perception/node_scripts/ssd_train_dataset.py
@@ -159,6 +159,8 @@ if __name__ == '__main__':
     p.add_argument("--val-iter", type=int, default=100)
     p.add_argument("--log-iter", type=int, default=10)
     p.add_argument("--model-iter", type=int, default=200)
+    p.add_argument("--single-process", action='store_true', default=False,
+                   help='run chainer using SerialIterator if this flag is set (default: False)')
 
     args = p.parse_args()
 
@@ -195,8 +197,13 @@ if __name__ == '__main__':
 
     train = TransformDataset(
         train, Transform(model.coder, model.insize, model.mean))
-    train_iter = chainer.iterators.MultiprocessIterator(
-        train, args.batchsize)
+
+    if args.single_process:
+        train_iter = chainer.iterators.SerialIterator(
+            train, args.batchsize)
+    else:
+        train_iter = chainer.iterators.MultiprocessIterator(
+            train, args.batchsize)
 
     test_iter = chainer.iterators.SerialIterator(
         test, args.batchsize,
